dbms/src/Core/Defines.h

```diff
@@ -109,3 +111,17 @@
 #if !defined(__x86_64__) && !defined(__aarch64__)
 //    #error PLATFORM_NOT_SUPPORTED
 #endif
+
+#define DISAGGREGATED_MODE_NONE "tiflash"
+#define DISAGGREGATED_MODE_STORAGE "tiflash_storage"
+#define DISAGGREGATED_MODE_COMPUTE "tiflash_compute"
+
+namespace DB
+{
+enum class DisaggregatedMode
+{
+    None,
+    Compute,
+    Storage,
+};
```

Comment:
Can we move to a separate file and put all the relevant methods in it?
Such as `getDisaggregatedMode`, etc